mlflow/langchain/api_request_parallel_processor.py

```diff
@@ -185,33 +210,43 @@ def call_api(
                             "invoke with the first value of the dictionary."
                         )
                         self.request_json = next(iter(self.request_json.values()))
+                        (
+                            prepared_request_json,
+                            did_perform_chat_conversion,
+                        ) = self._prepare_request_for_runnable_or_chain_inference(self.request_json)
+
                         response = self.lc_model.invoke(
-                            self._prepare_request_for_runnable_or_chain_inference(
-                                self.request_json
-                            ),
-                            config={"callbacks": callback_handlers},
+                            prepared_request_json, config={"callbacks": callback_handlers}
                         )
                 elif isinstance(self.request_json, list) and isinstance(
                     self.lc_model, runnables_supports_batch_types()
                 ):
                     response = self.lc_model.batch(
-                        self._prepare_request_for_runnable_or_chain_inference(self.request_json),
-                        config={"callbacks": callback_handlers},
+                        prepared_request_json, config={"callbacks": callback_handlers}
                     )
                 else:
                     response = self.lc_model.invoke(
-                        self._prepare_request_for_runnable_or_chain_inference(self.request_json),
-                        config={"callbacks": callback_handlers},
+                        prepared_request_json, config={"callbacks": callback_handlers}
                     )
+
+                if did_perform_chat_conversion:
+                    response = APIRequest._try_transform_response_to_chat_format(response)
             else:
+                (
+                    prepared_request_json,
+                    did_perform_chat_conversion,
+                ) = self._prepare_request_for_runnable_or_chain_inference(self.request_json)
                 response = self.lc_model(
-                    self._prepare_request_for_runnable_or_chain_inference(self.request_json),
+                    prepared_request_json,
                     return_only_outputs=True,
                     callbacks=callback_handlers,
                 )
 
-                # to maintain existing code, single output chains will still return only the result
-                if len(response) == 1:
+                if did_perform_chat_conversion:
```

Comment:
A random thought: Do people always want the converted response? Should we add a parameter to control whether to convert the response or not 🤔